dsgrid/config/dimension_association_manager.py

```diff
@@ -0,0 +1,93 @@
+"""
+Caches project dimension association tables in the Spark warehouse.
+One table per project can be stored. That table gets deleted whenever a project update
+invalidates it.
+
+The purpose is to allow dataset submissions to reuse existing tables.
+"""
+
+
+import logging
+
+import pyspark
+
+from dsgrid.utils.spark import (
+    try_load_stored_table,
+    save_table,
+    list_tables,
+    drop_table,
+)
+from dsgrid.utils.timing import timer_stats_collector, track_timing
+
+
+logger = logging.getLogger(__name__)
+
+_ASSOCIATIONS_DATA_TABLE = "dimension_associations_data"
+
+
+def try_load_dimension_associations(
+    project_id, dataset_id, pivoted_dimension
+) -> pyspark.sql.DataFrame | None:
+    """Load a project's dimension associations table if it exists.
+
+    Parameters
+    ----------
+    project_id : str
+    dataset_id : str
+    pivoted_dimension : DimensionType | None
+
+    Returns
+    -------
+    pyspark.sql.DataFrame | None
+
+    """
+    return try_load_stored_table(
+        _make_dimension_associations_table_name(project_id, dataset_id, pivoted_dimension)
+    )
```

Comment:
Is it correct that we used to have a project-wide association table, but now we only have dataset-specific association tables? I guess it is a simple matter to concatenate the dataset-specific association tables with the same pivoted dimension to get a project-wide association table?

Reply:
We used to build a project-wide table but then filter it by a data source before evaluating it, _I think_. I don't think we have a need for a project-wide table, but please correct me if I'm wrong.

Reply:
I don't think we immediately need a project-wide table, and maybe it's never a good idea to construct the whole thing